frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveImage/index.tsx

```diff
@@ -1,74 +1,112 @@
+    return (
+      !isScreenshotJank && !isNoScript && !isNonTissueChart && !isNonTissueYAxis
+    );
+  };
 
-const screenshotFilter = (domNode: HTMLElement): boolean => {
-  return (
-    !domNode.classList?.contains(EXCLUDE_IN_SCREENSHOT_CLASS_NAME) &&
-    domNode.tagName !== "NOSCRIPT"
-  );
-};
-
-const DownloadButton = styled(BPButton)`
-  margin-right: ${PT_GRID_SIZE_PX}px;
-`;
+function base64URLToArrayBuffer(url: string) {
+  // parse dataURL to base64 string
+  const base64 = url.split(",")[1];
+  const binary_string = window.atob(base64);
+  const len = binary_string.length;
+  const bytes = new Uint8Array(len);
+  for (let i = 0; i < len; i++) {
+    bytes[i] = binary_string.charCodeAt(i);
+  }
+  return bytes.buffer;
+}
 
 export default function SaveImage({
   selectedTissues,
   selectedGenes,
+  selectedCellTypes,
 }: {
   selectedTissues: Array<string>;
   selectedGenes: Array<string>;
+  selectedCellTypes: { [tissue: string]: CellType[] };
 }): JSX.Element {
   const [isOpen, setIsOpen] = useState(false);
   const [fileType, setFileType] = useState<"png" | "svg">("png");
   const handleButtonClick = useCallback(() => {
     setIsOpen(!isOpen);
   }, [isOpen]);
 
-  const saveImage = useCallback(async () => {
+  const handleDownload = useCallback(async () => {
     try {
       const heatmapNode = document.getElementById("view") as HTMLCanvasElement;
       heatmapNode.classList.add("CLONED");
-      const convertHTMLtoImage = fileType === "png" ? toPng : toSvg;
-      const imageURL = await convertHTMLtoImage(heatmapNode, {
-        backgroundColor: "white",
-        clonedClassName: "CLONED",
-        filter: screenshotFilter,
-        style: {
-          width: "min-content",
-        },
-      });
-      heatmapNode.classList.remove("CLONED");
-      download(
-        imageURL,
-        `CELLxGENE_gene_expression_${selectedTissues[0]}.${fileType}`
+      const isPNG = fileType === "png";
+      const convertHTMLtoImage = isPNG ? toPng : toSvg;
+      const images = await Promise.all(
+        selectedTissues.map(async (tissue) => {
+          const imageURL = await convertHTMLtoImage(heatmapNode, {
+            backgroundColor: "white",
+            filter: screenshotFilter(tissue),
+            height: getHeatmapHeight(selectedCellTypes[tissue]) + 200,
+            pixelRatio: 2,
+            width: getHeatmapWidth(selectedGenes) + 200,
+          });
+
+          return {
+            input: isPNG
+              ? base64URLToArrayBuffer(imageURL)
+              : decodeURIComponent(imageURL.split(",")[1]),
+            name: `${tissue}.${fileType}`,
+          };
+        })
       );
+      heatmapNode.classList.remove("CLONED");
+      const { downloadZip } = await import("client-zip");
+      const blob = await downloadZip(images).blob();
+      const link = document.createElement("a");
+      link.href = URL.createObjectURL(blob);
+      link.download = `CELLxGENE_gene_expression.zip`;
```

Comment:
If only one tissue, I think we should download the image itself instead of zip file.